src/metrics/rules/active_ruleset.go

```diff
@@ -310,28 +332,31 @@ func (as *activeRuleSet) rollupResultsFor(id []byte, timeNanos int64, matchOpts
 // nolint: unparam
 func (as *activeRuleSet) toRollupResults(
 	id []byte,
+	timeNanos int64,
 	cutoverNanos int64,
 	targets []rollupTarget,
 	keepOriginal bool,
 	tags [][]models.Tag,
-	matchOpts MatchOptions,
-) (rollupResults, error) {
+	matchOpts *MatchOptions,
+) error {
 	if len(targets) == 0 {
-		return rollupResults{}, nil
+		return nil
 	}
 
 	// If we cannot extract tags from the id, this is likely an invalid
 	// metric and we bail early.
 	_, sortedTagPairBytes, err := matchOpts.NameAndTagsFn(id)
 	if err != nil {
-		return rollupResults{}, err
+		return err
 	}
 
 	var (
-		multiErr           = xerrors.NewMultiError()
-		pipelines          = make([]metadata.PipelineMetadata, 0, len(targets))
-		newRollupIDResults = make([]idWithMatchResults, 0, len(targets))
-		tagPairs           []metricid.TagPair
+		multiErr    = xerrors.NewMultiError()
+		tagPairs    []metricid.TagPair
+		matchResult = matchOpts.MatchResult
+		// merge into forExistingID StagedMetadata created while processing mapping rules
+		feIdx              = len(matchResult.forExistingID) - 1
```

Comment:
Note that any updates here will be into the StagedMetadata we just created in the mapping rule section